### PR TITLE
Make airflow optional as it is not available in the test venue.

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary

Minor update for an edge case in test

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: __70 %

## ⚙️ Test Data and/or Report

Not tested, I trust AI. the behavior will be validated in the Test deployment.

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: terraform/iam.tf [hunk 1]
```diff
@@ -151,6 +151,7 @@ resource "aws_iam_policy" "mwaa_ecs_passrole_policy" {
 
 
 resource "aws_iam_role_policy_attachment" "mwaa_ecs_passrole" {
+  count      = var.mwaa_execution_role_name != "" ? 1 : 0
```
Comment (nutjob4life) [inline terraform/iam.tf]: I think `aws_iam_policy.mwaa_ecs_passrole_policy` is always created, but only the attachment is conditional. In test, Terraform will still create `pds-registry-sweeper-mwaa-ecs-passrole-policy` even though nothing attaches it.

Consider gating both resources with the same condition, for example:
```hcl
locals {
  enable_mwaa_passrole = var.mwaa_execution_role_name != ""
}
resource "aws_iam_policy" "mwaa_ecs_passrole_policy" {
  count = local.enable_mwaa_passrole ? 1 : 0
  # ...
}
resource "aws_iam_role_policy_attachment" "mwaa_ecs_passrole" {
  count      = local.enable_mwaa_passrole ? 1 : 0
  role       = var.mwaa_execution_role_name
  policy_arn = aws_iam_policy.mwaa_ecs_passrole_policy[0].arn
}
```
That keeps test from accumulating orphaned IAM policies.

File: terraform/variables.tf [hunk 2]
```diff
@@ -40,6 +40,7 @@ variable "permissions_boundary_policy_name" {
 variable "mwaa_execution_role_name" {
   type        = string
   description = "Name of the MWAA execution role that needs iam:PassRole to launch ECS tasks"
+  default = ""
```
Comment (nutjob4life) [inline terraform/variables.tf]: Maybe run `terraform fmt`? The `=` sign here is misaligned.

**Review by nutjob4life — CHANGES_REQUESTED** [review]
This looks great! Targets the right failure point, using `count` is standard Terraform practice. I'm sooooo clooooose to approving, but I did have two issues, one major and one minor. See the interspersed comments.

**Comment by jordanpadams** [comment]
@tloubrieu-jpl can you take a look at @nutjob4life's comments on this PR?

File: terraform/iam.tf (hunk 1 above)
Comment (tloubrieu-jpl) [reply terraform/iam.tf]: Thanks @nutjob4life , I fixed that.

File: terraform/variables.tf (hunk 2 above)
Comment (tloubrieu-jpl) [reply terraform/variables.tf]: Great idea, I did not know this command.

**Comment by tloubrieu-jpl** [comment]
@jordanpadams @nutjob4life , I made the proposed updates, ready for re-review. Thanks

**Review by jordanpadams — APPROVED** [review]

**Review by nutjob4life — APPROVED** [review]
Thanks for incorporating all that feedback. Approved.
